src/app/info/WikiItems.test.tsx

```diff
@@ -0,0 +1,281 @@
+});
+
+const mScrollIntoViewMock = jest.fn();
+window.HTMLElement.prototype.scrollIntoView = mScrollIntoViewMock;
+
+describe("Wiki items component", () => {
+    let mockData: DbWikiRow[];
+    beforeEach(() => {
+        cleanup();
+        jest.spyOn(window, "confirm").mockImplementation(() => true);
+
+        mockData = [
+            {
+                title: "Test 1",
+                content: "Test content 1",
+                wiki_key: "058049b5-7a7f-4f81-bf56-6dc9654e5a40",
+                row_order: 1,
+            },
+            {
+                title: "Test 2",
+                content: "Test content 2",
+                wiki_key: "731280a7-eb99-4229-aa49-84dbb112641c",
+                row_order: 2,
+            },
+        ];
+    });
+
+    it("renders the add, edit, reorder buttons for admins in display mode", () => {
+        render(
+            <StyleManager>
+                <RoleUpdateContext.Provider value={{ role: "admin", setRole: jest.fn() }}>
+                    <WikiItems rows={mockData} />
+                </RoleUpdateContext.Provider>
+            </StyleManager>
+        );
+        expect(
+            screen.getByRole("button", {
+                name: "+ Add",
+            })
+        );
+        expect(screen.getByTestId("#edit-1"));
+        expect(screen.getByTestId("#swap-up-1"));
+        expect(screen.getByTestId("#swap-down-2"));
+    });
+
+    it("has edit button for manager users, and on click, has update and delete buttons", () => {
+        render(
+            <StyleManager>
+                <RoleUpdateContext.Provider value={{ role: "manager", setRole: jest.fn() }}>
+                    <WikiItems rows={mockData} />
+                </RoleUpdateContext.Provider>
+            </StyleManager>
+        );
+        fireEvent.click(screen.getByTestId("#edit-1"));
+        expect(screen.getByTestId("#cancel-1"));
+        expect(screen.getByTestId("#update-1"));
+        expect(screen.getByTestId("#delete-1"));
+        expect(screen.getByTestId("#swap-up-1"));
+        expect(screen.getByTestId("#swap-down-1"));
+    });
+
+    it("does not render the add, reorder, or edit buttons for volunteers", () => {
+        render(
+            <StyleManager>
+                <RoleUpdateContext.Provider value={{ role: "volunteer", setRole: jest.fn() }}>
+                    <WikiItems rows={mockData} />
+                </RoleUpdateContext.Provider>
+            </StyleManager>
+        );
+        expect(
+            screen.queryByRole("button", {
+                name: "+ Add",
+            })
+        ).toBeNull();
+        expect(screen.queryByTestId("#swap-up-1")).toBeNull();
+        expect(screen.queryByTestId("#swap-down-2")).toBeNull();
+        expect(screen.queryByTestId("#edit-1")).toBeNull();
+    });
+
+    it("renders the wiki rows", () => {
+        render(
+            <StyleManager>
+                <RoleUpdateContext.Provider value={{ role: "admin", setRole: jest.fn() }}>
```

Comment:
Is there a way to test this for all roles, not just admins?

Reply:
rendering wiki rows doesn't depend on role, but we have changed this test to test with a volunteer role, as it has the lowest permissions. we've also duplicated all admin / manager tests to test both.